qemu/cpus.c

```diff
@@ -191,15 +169,10 @@ static int qemu_tcg_init_vcpu(CPUState *cpu)
         uc->tcg_halt_cond = cpu->halt_cond;
         snprintf(thread_name, VCPU_THREAD_NAME_SIZE, "CPU %d/TCG",
                 cpu->cpu_index);
-        if (qemu_thread_create(uc, cpu->thread, thread_name, qemu_tcg_cpu_thread_fn,
-                cpu, QEMU_THREAD_JOINABLE))
-            return -1;
+        qemu_thread_get_self(uc, cpu->thread);
```

Comment:
Yes, I think that once qemu_thread_create is remove, unicorn is locked into single cpu mode as there is no way that I see currently for cpus to yield to one another lacking threads. So there is a dilema: with threads, callbacks won't happen on the emu_start thread, and without threads, I think a single unicorn instance can't do multiple CPUs. Currently unicorn is hard coded for one cpu. Future uses of unicorn should be considered. My personal need is for a single CPU instance with a preference for callbacks on the emu_start thread.